File: src/constants/passportContent.ts
import type { ImageSourcePropType } from "react-native";

export type StampDefinition = {
  id: string;
  title: string;
  rarity: "common" | "rare" | "epic" | "legendary";
  issuedAt: string;
  description: string;
  source: string;
};

export type AchievementDefinition = {
  id: string;
  title: string;
  purpose: string;
  xp: number;
  missable: boolean;
  verification: string;
};

export type VisaDefinition = {
  id: string;
  title: string;
  neighborhood: string;
  grantedAt: string;
  requirement: string;
  description: string;
  accent: string;
};

export type BuildingDetail = {
  id: string;
  name: string;
  address: string;
  style: string;
  year: string;
  summary: string;
  image?: ImageSourcePropType;
  architect?: string;
  materials?: string;
  use?: string;
  type?: string;
  description?: string;
};

export type PassportListDefinition = {
  id: string;
  name: string;
  tagline: string;
  prompt: string;
  mood: string;
  buildings: BuildingDetail[];
};

export const stampCollection: StampDefinition[] = [
  {
    id: "flatiron-first",
    title: "Flatiron First Scan",
    rarity: "common",
    issuedAt: "2024-06-21",
    description:
      "Building Stamp • Awarded for scanning the Flatiron Building for the first time.",
    source: "Building scan",
  },
  {
    id: "quest-midtown",
    title: "Quest: Midtown Marvels",
    rarity: "rare",
    issuedAt: "2024-08-05",
    description:
      "Quest Stamp • Completed every Midtown Marvels clue before the quest timed out.",
    source: "Quest completion",
  },
  {
    id: "achievement-first-scan",
    title: "Achievement: First Scan",
    rarity: "epic",
    issuedAt: "2023-12-11",
    description:
      "Achievement Stamp • Marked your very first building scan in the city.",
    source: "Achievement unlock",
  },
  {
    id: "legendary-ticker",
    title: "Ticker Tape Legend",
    rarity: "legendary",
    issuedAt: "2024-02-19",
    description:
      "Legendary Stamp • Selected by the atelier jury during the Winter Ledger showcase.",
    source: "Curated showcase",
  },
  {
    id: "quest-dumbo",
    title: "Quest: DUMBO Steel",
    rarity: "rare",
    issuedAt: "2024-04-15",
    description:
      "Quest Stamp • Collected during the limited-run DUMBO Steel field quest.",
    source: "Quest completion",
  },
  {
    id: "oracle-aia",
    title: "AIA Open House",
    rarity: "common",
    issuedAt: "2024-10-08",
    description:
      "Building Stamp • Earned for checking into the AIA Open House tour checkpoint.",
    source: "Event check-in",
  },
  {
    id: "achievement-perfect-week",
    title: "Achievement: Perfect Week",
    rarity: "epic",
    issuedAt: "2024-05-03",
    description: "Achievement Stamp • 7 consecutive daily quest completions.",
    source: "Achievement unlock",
  },
  {
    id: "legendary-collection-master",
    title: "Collection Master",
    rarity: "legendary",
    issuedAt: "2024-06-30",
    description:
      "Legendary Stamp • Granted for completing every archival quest variant in a chapter.",
    source: "Collection mastery",
  },
];

export const achievementLedger: AchievementDefinition[] = [
  {
    id: "first_scan",
    title: "First Scan",
    purpose: "Scan your first building to prove you’ve started exploring.",
    xp: 25,
    missable: false,
    verification:
      "Unlocks once any building scan is recorded for your profile.",
  },
  {
    id: "100_scans",
    title: "100 Unique Scans",
    purpose: "Encourages breadth—visit 100 distinct buildings.",
    xp: 500,
    missable: false,
    verification:
      "Checks for 100 distinct building scans logged under your account.",
  },
  {
    id: "style_explorer_modern",
    title: "Modern Explorer",
    purpose: "Visit ten Modernist buildings to understand the style in depth.",
    xp: 200,
    missable: false,
    verification:
      "Validates 10 Modern-style buildings scanned via the architectural dataset.",
  },
  {
    id: "dedicated_traveler",
    title: "Dedicated Traveler",
    purpose: "Complete a single derive or jink that lasts at least 90 minutes.",
    xp: 200,
    missable: false,
    verification: "Looks for a derive session with duration >= 90 minutes.",
  },
  {
    id: "perfect_week",
    title: "Perfect Week",
    purpose: "Seven daily quests in a row—time-boxed and missable.",
    xp: 1200,
    missable: true,
    verification: "Requires a seven-day streak of completed daily quests.",
  },
  {
    id: "dawn_scanner",
    title: "Dawn Scanner",
    purpose: "Finish a daily quest before 08:00 local time.",
    xp: 150,
    missable: true,
    verification:
      "Confirms a completed quest with completion time before 08:00.",
  },
  {
    id: "collection_champion",
    title: "Collection Champion",
    purpose:
      "Collect every quest stamp in a complete collection—extremely rare.",
    xp: 10000,
    missable: true,
    verification:
      "Validates ownership of all quest variant stamps in a collection.",
  },
  {
    id: "first_visa",
    title: "First Visa",
    purpose: "Earn your first neighborhood visa and cement local credentials.",
    xp: 100,
    missable: false,
    verification:
      "Granted once any entry exists in user_visas for your account.",
  },
];

export const visaCarousel: VisaDefinition[] = [
  {
    id: "midtown",
    title: "Midtown Marvels Visa",
    neighborhood: "Midtown",
    grantedAt: "2024-07-12",
    requirement: "Scan 12 landmark towers between 34th Street and 59th Street.",
    description:
      "Issued after you walked the Midtown canyon, capturing deco crowns and glass spires.",
    accent: "#2D5B91",
  },
  {
    id: "bedstuy",
    title: "Bed-Stuy Brownstone Visa",
    neighborhood: "Bedford–Stuyvesant",
    grantedAt: "2024-05-28",
    requirement:
      "Visit 10 distinct stoops in the Stuyvesant Heights landmark district.",
    description:
      "Awarded for knowing the blocks by heart—detailing lintels, cornices, and backyard stories.",
    accent: "#8B4A2D",
  },
  {
    id: "lic",
    title: "LIC Waterfront Visa",
    neighborhood: "Long Island City",
    grantedAt: "2024-09-03",
    requirement: "Complete three twilight derives along the Queens waterfront.",
    description:
      "You traced the gantries and silos at dusk—documenting rail relics and neon glow.",
    accent: "#1F8A70",
  },
];

export const passportLists: PassportListDefinition[] = [
  {
    id: "deco-giants",
    name: "Deco Giants",
    tagline: "Where steel meets style.",
    prompt: "Add to list!",
    mood: "Polished chrome, sunrise reflections, and lobby murals.",
    buildings: [
      {
        id: "chrysler",
        name: "Chrysler Building",
        address: "405 Lexington Ave",
        style: "Art Deco",
        year: "1930",
        summary:
          "Iconic crown with radiating metal arches and automotive gargoyles.",
      },
      {
        id: "ge",
        name: "570 Lexington",
        address: "570 Lexington Ave",
        style: "Art Deco",
        year: "1931",
        summary: "Red brick tower with limestone tracery and spire lantern.",
      },
      {
        id: "30rock",
        name: "30 Rockefeller Plaza",
        address: "45 Rockefeller Plaza",
        style: "Art Deco",
        year: "1933",
        summary:
          "Massed limestone with sculpted reliefs and soaring observation deck.",
      },
      {
        id: "citybank",
        name: "City Bank-Farmers Trust",
        address: "20 Exchange Pl",
        style: "Art Deco",
        year: "1931",
        summary:
          "Tiered ziggurat silhouette with bronze doors and stylized eagles.",
      },
      {
        id: "walker",
        name: "Walker Tower",
        address: "212 W 18th St",
        style: "Art Deco",
        year: "1929",
        summary: "Chelsea telegraph HQ reborn with terracotta chevron panels.",
      },
    ],
  },
  {
    id: "industrial-poetics",
    name: "Industrial Poetics",
    tagline: "Factories that hum with rhythm.",
    prompt: "Add to list!",
    mood: "Riveted beams, brick soot, mechanical grace notes.",
    buildings: [
      {
        id: "dumbo-warehouse",
        name: "Empire Stores",
        address: "55 Water St",
        style: "Warehouse Revival",
        year: "1869",
        summary:
          "Civil War coffee warehouse—brick vaults with Brooklyn Bridge framed.",
      },
      {
        id: "neon-pepsi",
        name: "Pepsi-Cola Sign",
        address: "4-09 47th Rd",
        style: "Industrial Neon",
        year: "1936",
        summary:
          "Floating red neon script anchored to a converted bottling plant.",
      },
      {
        id: "knitting-factory",
        name: "Brooklyn Navy Yard - Building 77",
        address: "141 Flushing Ave",
        style: "Streamline Moderne",
        year: "1942",
        summary:
          "Re-skinned supply depot with stepped glass curtain and roof farm.",
      },
      {
        id: "sugar",
        name: "Domino Sugar Refinery",
        address: "292 Kent Ave",
        style: "Romanesque Industrial",
        year: "1884",
        summary: "Landmark brick refinery reimagined with glass barrel insert.",
      },
    ],
  },
  {
    id: "brutalist-beacons",
    name: "Brutalist Beacons",
    tagline: "Concrete poetry of the late modern city.",
    prompt: "Add to list!",
    mood: "Board-formed concrete, heroic stairs, softened with sky gardens.",
    buildings: [
      {
        id: "met-breuer",
        name: "The Met Breuer",
        address: "945 Madison Ave",
        style: "Brutalist",
        year: "1966",
        summary:
          "Marcel Breuer’s inverted ziggurat with bush-hammered concrete façade.",
      },
      {
        id: "lincoln-center",
        name: "Lincoln Center Library",
        address: "40 Lincoln Center Plaza",
        style: "Brutalist",
        year: "1965",
        summary: "Low-slung mass with slender light wells and recessed plaza.",
      },
      {
        id: "hunter",
        name: "Hunter College Library",
        address: "695 Park Ave",
        style: "Brutalist",
        year: "1984",
        summary: "Angular tower carved by terraces and raw concrete fins.",
      },
      {
        id: "lehman",
        name: "Lehman College Art Gallery",
        address: "250 Bedford Park Blvd W",
        style: "Brutalist",
        year: "1968",
        summary:
          "Cantilevered mass hovering over a reflecting pool with ribbed soffits.",
      },
    ],
  },
  {
    id: "seaside-escape",
    name: "Seaside Escape",
    tagline: "Buildings tuned to the Atlantic breeze.",
    prompt: "Add to list!",
    mood: "Bleached concrete, salt-crusted steel, and cooling canopies.",
    buildings: [
      {
        id: "rockaways",
        name: "Rockaway Beach Skate Pavilion",
        address: "94-00 Shore Front Pkwy",
        style: "Coastal Modern",
        year: "2017",
        summary:
          "Wave-like canopy in galvanized steel with skate ribbon below.",
      },
      {
        id: "coney",
        name: "Coney Island Steeplechase Plaza",
        address: "1904 Surf Ave",
        style: "Adaptive Reuse",
        year: "2013",
        summary:
          "Historic parachute jump framed with new timber decks and views.",
      },
      {
        id: "sunset-harbor",
        name: "Sunset Cove Nature Center",
        address: "Bayview Ave, Broad Channel",
        style: "Sustainable",
        year: "2020",
        summary:
          "Resilient wetlands pavilion with passive ventilation corridors.",
      },
      {
        id: "st-george",
        name: "Empire Outlets Canopy",
        address: "55 Richmond Ter",
        style: "Contemporary",
        year: "2019",
        summary: "Sculpted canopy capturing harbor winds with LED nightwash.",
      },
    ],
  },
];

File: JinkiOS/jink/JinkApp/Data/PassportContent.swift
import Foundation
import CoreLocation

// MARK: - Stamp Definitions

struct StampDefinition: Identifiable {
    let id: String
    let slug: String
    let title: String
    let description: String
    let rarity: StampRarity
    let series: String
}

enum StampRarity: String {
    case common, rare, epic, legendary

    var displayName: String {
        rawValue.capitalized
    }
}

// MARK: - Achievement Definitions

struct AchievementDefinition: Identifiable {
    let id: String
    let title: String
    let description: String
    let xpReward: Int
    let icon: String
    let isMissable: Bool
    let verificationText: String?
}


// MARK: - Passport List Definitions

struct PassportListBuilding: Identifiable {
    let id: String
    let bin: String
    let name: String
    let address: String
    let latitude: Double
    let longitude: Double
}

struct PassportListDefinition: Identifiable {
    let id: String
    let name: String
    let tagline: String
    let mood: String
    let buildings: [PassportListBuilding]
}

// MARK: - Tour Definitions

enum CheckpointType: String {
    case building, waypoint, viewpoint
}

struct TourCheckpoint: Identifiable {
    let id: String
    let type: CheckpointType
    let name: String
    let latitude: Double
    let longitude: Double
    let narrative: String
    let funFact: String
    let action: String
    let bin: String?
    let address: String?
}

struct TourDefinition: Identifiable, Hashable {
    static func == (lhs: TourDefinition, rhs: TourDefinition) -> Bool { lhs.id == rhs.id }
    func hash(into hasher: inout Hasher) { hasher.combine(id) }
    let id: String
    let name: String
    let subtitle: String
    let difficulty: String
    let durationMinutes: Int
    let distanceMiles: Double
    let xpReward: Int
    let badgeId: String
    let badgeName: String
    let checkpoints: [TourCheckpoint]
    let summaryText: String
}

// MARK: - Static Content

enum PassportContent {

    // MARK: Stamps
    static let stampCollection: [StampDefinition] = [
        StampDefinition(
            id: "flatiron_first_scan",
            slug: "flatiron_first_scan",
            title: "Flatiron First Scan",
            description: "You scanned the iconic Flatiron Building — a true NYC landmark.",
            rarity: .common,
            series: "building"
        ),
        StampDefinition(
            id: "quest_midtown_marvels",
            slug: "quest_midtown_marvels",
            title: "Quest: Midtown Marvels",
            description: "Completed the Midtown Marvels quest and earned your place in history.",
            rarity: .rare,
            series: "quest"
        ),
        StampDefinition(
            id: "achievement_first_scan",
            slug: "achievement_first_scan",
            title: "Achievement: First Scan",
            description: "You scanned your very first building. The city is your oyster.",
            rarity: .common,
            series: "achievement"
        ),
        StampDefinition(
            id: "ticker_tape_legend",
            slug: "ticker_tape_legend",
            title: "Ticker Tape Legend",
            description: "You walked the historic route of NYC's ticker-tape parade corridor.",
            rarity: .epic,
            series: "building"
        ),
        StampDefinition(
            id: "quest_dumbo_steel",
            slug: "quest_dumbo_steel",
            title: "Quest: DUMBO Steel",
            description: "Navigated DUMBO's industrial waterfront and lived to tell the tale.",
            rarity: .rare,
            series: "quest"
        ),
        StampDefinition(
            id: "aia_open_house",
            slug: "aia_open_house",
            title: "AIA Open House",
            description: "Attended AIA Open House NYC and explored behind closed doors.",
            rarity: .epic,
            series: "building"
        ),
        StampDefinition(
            id: "achievement_perfect_week",
            slug: "achievement_perfect_week",
            title: "Achievement: Perfect Week",
            description: "Scanned buildings every day for a full week. Dedication personified.",
            rarity: .legendary,
            series: "achievement"
        ),
        StampDefinition(
            id: "collection_master",
            slug: "collection_master",
            title: "Collection Master",
            description: "Collected every stamp in a complete series. Legendary explorer.",
            rarity: .legendary,
            series: "achievement"
        ),
    ]

    // MARK: Achievements
    static let achievementLedger: [AchievementDefinition] = [
        AchievementDefinition(
            id: "first_scan",
            title: "First Scan",
            description: "Scan your first building",
            xpReward: 25,
            icon: "camera.viewfinder",
            isMissable: false,
            verificationText: "Automatically verified on first scan"
        ),
        AchievementDefinition(
            id: "100_scans",
            title: "Century Scanner",
            description: "Scan 100 buildings",
            xpReward: 500,
            icon: "building.2.fill",
            isMissable: false,
            verificationText: "Verified by scan count"
        ),
        AchievementDefinition(
            id: "style_explorer_modern",
            title: "Style Explorer: Modern",
            description: "Scan 10 modernist buildings",
            xpReward: 200,
            icon: "square.stack.3d.up.fill",
            isMissable: false,
            verificationText: "Verified by archetype scans"
        ),
        AchievementDefinition(
            id: "dedicated_traveler",
            title: "Dedicated Traveler",
            description: "Complete 5 walks",
            xpReward: 200,
            icon: "figure.walk",
            isMissable: false,
            verificationText: "Verified by walk count"
        ),
        AchievementDefinition(
            id: "perfect_week",
            title: "Perfect Week",
            description: "Scan every day for 7 days in a row",
            xpReward: 1200,
            icon: "calendar.badge.checkmark",
            isMissable: true,
            verificationText: "Must maintain 7-day streak"
        ),
        AchievementDefinition(
            id: "dawn_scanner",
            title: "Dawn Scanner",
            description: "Scan a building before 7am",
            xpReward: 150,
            icon: "sunrise.fill",
            isMissable: true,
            verificationText: "Verified by scan timestamp"
        ),
        AchievementDefinition(
            id: "collection_champion",
            title: "Collection Champion",
            description: "Complete all stamps in a series",
            xpReward: 10000,
            icon: "star.circle.fill",
            isMissable: true,
            verificationText: "Verified by stamp completion"
        ),
    ]


    // MARK: Passport Lists
    static let passportLists: [PassportListDefinition] = [
        PassportListDefinition(
            id: "deco_giants",
            name: "Deco Giants",
            tagline: "The gilded peaks of Manhattan",
            mood: "Opulent",
            buildings: [
                PassportListBuilding(id: "chrysler", bin: "1001830", name: "Chrysler Building", address: "405 Lexington Ave", latitude: 40.7516, longitude: -73.9755),
                PassportListBuilding(id: "empire_state", bin: "1001028", name: "Empire State Building", address: "350 Fifth Ave", latitude: 40.7484, longitude: -73.9857),
                PassportListBuilding(id: "rockefeller", bin: "1001830", name: "30 Rockefeller Plaza", address: "30 Rockefeller Plaza", latitude: 40.7587, longitude: -73.9787),
                PassportListBuilding(id: "one_wall_st", bin: "1000484", name: "One Wall Street", address: "1 Wall St", latitude: 40.7071, longitude: -74.0121),
                PassportListBuilding(id: "woolworth", bin: "1001831", name: "Woolworth Building", address: "233 Broadway", latitude: 40.7128, longitude: -74.0080),
            ]
        ),
        PassportListDefinition(
            id: "industrial_poetics",
            name: "Industrial Poetics",
            tagline: "Where iron meets sky",
            mood: "Raw",
            buildings: [
                PassportListBuilding(id: "high_line_1", bin: "1008882", name: "Standard High Line", address: "848 Washington St", latitude: 40.7393, longitude: -74.0084),
                PassportListBuilding(id: "chelsea_market", bin: "1008390", name: "Chelsea Market", address: "75 9th Ave", latitude: 40.7424, longitude: -74.0044),
                PassportListBuilding(id: "domino", bin: "3007685", name: "Domino Sugar Factory", address: "1 S 1st St, Brooklyn", latitude: 40.7140, longitude: -73.9625),
                PassportListBuilding(id: "brooklyn_navy", bin: "3050000", name: "Brooklyn Navy Yard", address: "63 Flushing Ave, Brooklyn", latitude: 40.6982, longitude: -73.9732),
            ]
        ),
        PassportListDefinition(
            id: "brutalist_beacons",
            name: "Brutalist Beacons",
            tagline: "Concrete poetry in motion",
            mood: "Bold",
            buildings: [
                PassportListBuilding(id: "ford_foundation", bin: "1001893", name: "Ford Foundation Building", address: "320 E 43rd St", latitude: 40.7505, longitude: -73.9739),
                PassportListBuilding(id: "bobst", bin: "1009847", name: "Bobst Library NYU", address: "70 Washington Square S", latitude: 40.7294, longitude: -73.9977),
                PassportListBuilding(id: "paul_hall", bin: "1009000", name: "Paul Hall Center", address: "17 Battery Pl", latitude: 40.7032, longitude: -74.0145),
                PassportListBuilding(id: "breuer", bin: "1001640", name: "Whitney Museum (Breuer)", address: "945 Madison Ave", latitude: 40.7729, longitude: -73.9634),
            ]
        ),
        PassportListDefinition(
            id: "seaside_escape",
            name: "Seaside Escape",
            tagline: "Where the city meets the water",
            mood: "Serene",
            buildings: [
                PassportListBuilding(id: "battery_park_1", bin: "1000001", name: "Castle Clinton", address: "Battery Park", latitude: 40.7034, longitude: -74.0170),
                PassportListBuilding(id: "pier17", bin: "1000002", name: "Pier 17", address: "89 South St", latitude: 40.7062, longitude: -74.0030),
                PassportListBuilding(id: "brooklyn_heights", bin: "3000100", name: "Brooklyn Heights Promenade", address: "Columbia Heights, Brooklyn", latitude: 40.6960, longitude: -73.9965),
                PassportListBuilding(id: "governors_island", bin: "5000100", name: "Fort Jay", address: "Governors Island", latitude: 40.6900, longitude: -74.0166),
            ]
        ),
    ]
}
